--- a/library/bignum.c
+++ b/library/bignum.c
@@ -1734,11 +1734,13 @@ cleanup:
     return ret;
 }

+#if !defined(MBEDTLS_MPI_EXP_MOD_ALT)
 int mbedtls_mpi_exp_mod(mbedtls_mpi *X, const mbedtls_mpi *A,
                         const mbedtls_mpi *E, const mbedtls_mpi *N,
                         mbedtls_mpi *prec_RR)
 {
     return mbedtls_mpi_exp_mod_optionally_safe(X, A, E, MBEDTLS_MPI_IS_SECRET, N, prec_RR);
 }
+#endif /* !MBEDTLS_MPI_EXP_MOD_ALT */

 int mbedtls_mpi_exp_mod_unsafe(mbedtls_mpi *X, const mbedtls_mpi *A,
                                const mbedtls_mpi *E, const mbedtls_mpi *N,
